### ISSUE TITLE
Default local preferences set "Use at most" to zero gigabytes

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1308]
**Reported by Ageless on 3 Sep 44141874 19:38 UTC**
Yesterday I installed BOINC clean on a system that never had a BOINC on it before. I used 7.2.39 - 64bit, that I had downloaded from 
http://boinc.berkeley.edu/download.php

I was testing something about the screen saver flickering, but during
my travels in BOINC, I found that the default value for Use at most is set to 0.00GB of disk space. See the attached image. 

This will mean that project servers will refuse to send work to these
systems, until that value has been adjusted to something normal.

Do know that a lot of people will look in BOINC's local preferences,
and exit out through the OK button, not knowing that this will
immediately use the new local preferences, overriding their web
preferences.

Since the Use at most X GB disk space value is a required one, can we
make sure that it can never be zero? Or when it's read as being zero,
that we use the value from the global_prefs.xml file? Or set it by
default to 5Gb for PCs and Macs (NOT on Android!).

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1337

**Comment by ChristianBeer** [comment]
Defaults were adjusted at least in current 7.6.33 the problem is not present anymore.

**Comment by Ageless93** [comment]
Are we sure about that? Because I see this problem happen regularly in the
forums.
Just three days ago in
https://boinc.berkeley.edu/dev/forum_thread.php?id=11555&postid=77232#77232
the user posted for something else, but his 7.6.33 shows:

17/04/07 20:04:07 | | Reading preferences override file
17/04/07 20:04:07 | | Preferences:
17/04/07 20:04:07 | | max memory usage when active: 16352.53MB
17/04/07 20:04:07 | | max memory usage when idle: 29434.55MB
17/04/07 20:04:07 | | max disk usage: 0.00GB

No one sets 0.00GB for disk space in the local preferences, so this must be
the default setting.


-- Jord van der Elst.

**Comment by ChristianBeer** [comment]
Might be. I tested it by installing a fresh Client/Manager and looking at the default local preference settings before attaching any project. One needs to track down all usages of the underlying variable I guess because there are many ways to set this to 0.

**Comment by CharlieFenton** [comment]
A look at the code shows that the value is set to 0 internally when the checkbox in the Advanced Preferences dialog is off, meaning that this value should be ignored. There are 3 possible settings for disk usage limits, and, as the dialog says, "BOINC will use the least restrictive of these settings."

If you look at the code on lines 66-69 in cs_prefs.cpp (in GIT master), the value of global_prefs.disk_max_used_gb is used only if it is non-zero. The text edit field in the Advanced Preferences dialog is blanked out when the checkbox is off so as not to confuse the user by showing 0. I believe that in earlier versions of this dialog, there was no checkbox but instead text saying "0 means no limit."

Similarly, the global_prefs.disk_min_free_gb value is set to 0 in the Advanced Preferences dialog when the checkbox is off but its text edit field is blanked out  so as not to confuse the user by showing 0. But the code in CLIENT_STATE::allowed_disk_usage(double boinc_total) returns 0 for allowed disk usage when both global_prefs.disk_max_used_gb and global_prefs.disk_min_free_gb are zero. A zero value again means "no limit".

So I believe the problem is on lines 668-671 in cs_prefs.cpp, which should not display the value if it is zero. or perhaps should say "no limit" in that case.

In fact, looking back at old correspondence, I found this from almost 3 years ago:

**Comment by CharlieFenton** [comment]
After writing the above, I realized that it is a bit more complicated than I was remembering. The values set in the Advanced Preferences dialog are written to the global_prefs_override.xml file. I believe that a zero value in this file simply means not to override the web-based prefs in the global_prefs.xml file (presumably because the checkbox was off in the dialog) or perhaps it means "continue to use the most recently set value."

But if the checkbox is off in the user's Computing Preferences page on the SETI@home web site, then those values in the global_prefs.xml file will be 0, which I still believe means "no limit."  By turning off all 3 checkboxes on the web page, telling the Manager to update the project, and selecting "Use web prefs" in the Manager's Advanced Preferences dialog, I was able to get the Event Log to display "max disk usage: 0.00GB"

So I still believe that the problem is on lines 668-671 in cs_prefs.cpp, which should not display the value if it is zero. or perhaps should say "no limit" in that case.
